src/cmd/controller/main.go

```diff
@@ -144,71 +147,32 @@ func main() {
 		logger.Fatal("failed to create EC2 client")
 	}
 
-	nodeGetter, err := node.NewGetter(kubeClient)
-	if err != nil {
-		logger.With("error", err).Fatal("failed to create node getter")
-	}
-	nodeNameGetter, err := nodename.NewGetter(ec2Client)
-	if err != nil {
-		logger.With("error", err).Fatal("failed to create node name getter")
-	}
-
-	asgTerminateEventV1Parser, err := asgterminateeventv1.NewParser(asgClient)
-	if err != nil {
-		logger.With("error", err).Fatal("failed to create ASG instance-terminate lifecycle event v1 parser")
-	}
-	asgTerminateEventV2Parser, err := asgterminateeventv2.NewParser(asgClient)
-	if err != nil {
-		logger.With("error", err).Fatal("failed to create ASG instance-terminate lifecycle event v2 parser")
-	}
-	sqsMessageParser, err := terminator.NewSqsMessageParser(event.NewParser(
-		asgTerminateEventV1Parser,
-		asgTerminateEventV2Parser,
-		rebalancerecommendationeventv0.NewParser(),
-		scheduledchangeeventv1.NewParser(),
-		spotinterruptioneventv1.NewParser(),
-		statechangeeventv1.NewParser(),
-	))
-	if err != nil {
-		logger.With("error", err).Fatal("failed to create SQS message parser")
-	}
-
-	terminatorGetter, err := terminator.NewGetter(kubeClient)
-	if err != nil {
-		logger.With("error", err).Fatal("failed to create terminator getter")
-	}
-
-	sqsMessageClient, err := sqsmessage.NewClient(sqsClient)
-	if err != nil {
-		logger.With("error", err).Fatal("failed to create SQS message client")
-	}
-	terminatorSqsClientBuilder, err := terminator.NewSqsClientBuilder(sqsMessageClient)
-	if err != nil {
-		logger.With("error", err).Fatal("failed to create terminator SQS message client builder")
-	}
-
-	cordonDrainerBuilder, err := kubectlcordondrainer.NewBuilder(
-		clientSet,
-		kubectlcordondrainer.DefaultCordoner,
-		kubectlcordondrainer.DefaultDrainer,
+	eventParser := event.NewAggregatedParser(
+		asgterminateeventv1.Parser{ASGLifecycleActionCompleter: asgClient},
```

Comment:
It corresponds to the value of the `"version"` field from the EventBridge message (stored in `AWSMetadata.Version`).  Each event type has a versioned schema defined in EventBridge, e.g. [here](https://github.com/aws/aws-node-termination-handler/pull/612/files#diff-544ce1b445bf7f98041c047ae1a9ce60896914a1c1d80f772bed0f4a8e60a378R26-R27).

Reply:
Ahh gotcha, I feel like I'm getting lost in the versions here: CRD/API version, app version, module/pkg version, and now EventBridge schema version.

With that said, I'm still leaning towards removing the `version` (v0, v1) from the project directories/file paths. I don't think the schema version needs to be exposed in the project, but instead handled within an event's `Parse` logic. For example with `asgterminate` event, `Parse` would try to unmarshal into `v1` struct first.. if that fails try `v2`. 
  * Another idea (related to my other comment) is a "super" struct `AsgTerminate` that is a superset of all fields, unmarshal into that, and then check field(s) for version.

Last concern with breaking out by schema version is if `v3` is released tomorrow with very minimal changes, then we need to create a new directory and set of files that would be mostly copy/paste which I don't think is necessary.

Reply:
> if v3 is released tomorrow *with very minimal changes* [emphasis mine]

It is also possible that new versions would include new and important data fields, or require new API calls.

I think the question is: what should NTH do if it encounters a newer version number of a known event type?

1. Ignore it
    * This is the currently implemented behavior; an error is logged
1. Handle it the same as the previous version
    * Should probably log a warning about the unknown version in case there is a failure later

Reply:
> I think the question is: what should NTH do if it encounters a newer version number of a known event type?

> 1. Ignore it
> This is the currently implemented behavior; an error is logged
> 2. Handle it the same as the previous version
Should probably log a warning about the unknown version in case there is a failure later

I'd go with number 2 and make it a best effort to handle it (or maybe configurable?). However, I think regardless of how we implement unknown version handling we don't need to expose the version at the package level.